wp-content/themes/csisnuclearnetwork/inc/template-tags.php

```diff
@@ -377,3 +386,53 @@ function nuclearnetwork_display_footnotes() {
 		}
 	}
 endif;
+
+/**
+ * Displays the post's type and subtypes.
+ *
+ *
+ * @return string $html The subtypes.
+ */
+if (! function_exists('nuclearnetwork_display_subtypes')) :
+	function nuclearnetwork_display_subtypes() {
+
+		
+		$post_type = get_post_type();
+		global $post;
+		
+		if ( $post_type === 'events' ) {
+			$post_type_name = 'Event';
+			$tax_name = 'event_types';
+			$slug = $post_type;
+		} elseif ($post_type === 'updates' ) {
+			$post_type_name = 'Program Update';
+			$tax_name = '';
+			$slug = 'updates';
+		} elseif ($post_type === 'programs' && is_single() ) {
+			$post_type_name = 'Program';
+			$tax_name = '';
+			$slug = $post_type;
+		} elseif ($post_type === 'post' ) {
+			$post_type_name = 'Analysis';
+			$tax_name = 'analysis_subtype';
+			$slug = 'analysis';
+		}
+		
+		echo '<div class="post-meta post-meta__terms"><a href="/' . $slug . '" class="post-meta__terms-type text--bold">' . $post_type_name . get_the_term_list( $post->ID, $tax_name, ' / </a>', ', ') . '</div>';
```

Comment:
I'm concerned about hardcoding these URLs and titles in to the theme like this - makes it a lot harder to change in the future if needed. Is there a way you'd be able to use the built in permalink functions here to simplify this? If not, then be sure to add a note to the README that this is a part of the theme that would need to be manually updated if they ever want to update the language for these content types.